src/App.jsx

```diff
@@ -1,24 +1,58 @@
-import logo from "./assets/technigo-logo.svg";
-// Import the custom hook useQRCodeGenerator
-import { QrExample } from "./components/QrExample";
+
+// import logo from "./assets/technigo-logo.svg";
+// import { QrExample } from "./components/QrExample";
+import { useQRCodeGenerator } from "./hooks/useQRCodeGenerator";
+import "animate.css";
+import Lottie from "lottie-react";
+import animationData from "./assets/Animation.json";
 
 // Define the App component
 export const App = () => {
   // Destructure variables, properties and methods from the useQRCodeGenerator hook that you imported above here :)
+  const {
+    url,
+    qr,
+    showInput,
+    setUrl,
+    generateQRCode,
+    downloadQRCode,
+    repeatAction,
+  } = useQRCodeGenerator();
 
   // Return the JSX to render the component
+
   return (
-    <div className="">
+    <div className="app">
       {/* Render the title */}
-      <img className="logo" src={logo} alt="" />
-      <h1>Technigo QR Code Generator</h1>
-      <p>Start Here</p>
-      <QrExample />
 
-      {/* Conditionally render based on wether the user is inputting an URL to generate a QR Code or the user wnats to downaload the generated QR Code from the url input */}
-      {/* {yourReactiveVariableThatTogglesTheDownloadQrCcodeOrInputField ? () : ()} */}
+      <Lottie className="lottie-animation" animationData={animationData} />
+      <h1 className="animate__fadeInDown">QR Code Generator</h1>
+
+      {showInput ? (
+        <div>
+          {/* Render input field and "Generate" button when showInput is true */}
+          <input
+            type="text"
+            placeholder="e.g. https://google.com"
+            value={url}
+            onChange={(e) => setUrl(e.target.value)}
+          />
+          <button onClick={generateQRCode}>Generate</button>
+        </div>
+      ) : (
+        <div className="qr-buttons">
+          {/* Render QR code image, "Download," and "Repeat" buttons when showInput is false */}
+          <img src={qr} alt="QR Code" />
+          <div className="buttons">
+            <button onClick={downloadQRCode}>Download</button>
+            <button onClick={repeatAction}>Repeat</button>
+          </div>
+        </div>
+      )}
+
```

Comment:
Should there be some kind of URL validation here? Any input generates a QR code.